client/src/components/Login/SignUpForm/index.js

```diff
@@ -59,11 +61,60 @@ const SignUpForm = ({ switchView }) => {
     }
   };
 
+  const validateUsername = () => {
+    let usernameState = {
+      error: false,
+      message: ""
+    };
+    if (!username) {
+      usernameState.error = true;
+      usernameState.message = "Username is required";
+    }
+    setValidationState({
+      ...validationState,
+      username: usernameState
+    });
+  };
+
+  const validatePassword = () => {
+    let passwordState = {
+      error: false,
+      message: ""
+    };
+    if (!password) {
+      passwordState.error = true;
+      passwordState.message = "Password is required.";
+    } else if (password.length < 8) {
+      passwordState.error = true;
+      passwordState.message = "Password must be at least 8 characters.";
+    }
+    setValidationState({
+      ...validationState,
+      password: passwordState
+    });
+  };
+
+  const signUp = async () => {
+    try {
+        const signUpStatus = await ServiceClient.signUp({ name, username, password });
+        if (signUpStatus.success) {
+            updateAuthToken && udpateAuthtoken(signUpStatus.authToken);
+            redirectToHome(); // ?
```

Comment:
The rediectToHome() method will switch the url to be "/" and it will render the home page of our app.